blink_detector.py

```diff
@@ -12,6 +15,100 @@ class EyeSide(Enum):
     RIGHT = auto()
 
 
+class DynamicThresholdMaker:
+    """Take a certain number of landmarks that contains face to dynamically
+    determine a threshold of eye aspect ratio.
+
+    It can be used with a BlinkDetector to provide better detections on different
+    users.
+    "Decimal" module is used to reduce round-off errors, provide better precision.
+    """
+    def __init__(
+            self,
+            temp_thres: Union[Decimal, float] = Decimal('0.24'),
+            num_thres: int = 100) -> None:
+        """
+        Arguments:
+            temp_thres:
+                The temporary threshold used before the dynamic threshold is ready,
+                which prevents from getting an unreliable threshold due to low
+                number of samples. Should be in range [0.15, 0.5].
+            num_thres:
+                Higher number (included) of samples than this is considered to
+                be reliable.
+        """
+        if temp_thres < 0.15 or temp_thres > 0.5:
+            raise ValueError("resonable threshold should >= 0.15 and <= 0.5")
+        if num_thres < 100:
+            raise ValueError("number of samples under 100 makes the normal eye"
+                             "aspect ratio susceptible to extreme values")
+
+        self._dyn_thres = Decimal(temp_thres)
+        self._num_thres = num_thres
+        # To calculate the mean and standard deviation without re-summing all the
+        # samples, we store the real time sum and sum of squares.
+        self._cur_sum = Decimal(0)
+        self._cur_sum_of_sq = Decimal(0)
+        # The only reason we stored the sample ratios is to remove the old
+        # ratios in order.
+        self._samp_ratios: Deque[Decimal] = deque()
+
+    @property
+    def threshold(self) -> Decimal:

```

Comment:
The docstring for a `@property` data descriptor should use the same style as the docstring for an attribute or a [function argument](https://google.github.io/styleguide/pyguide.html#doc-function-args) (`"""The dynamic threshold..."""`, rather than `"""Returns the dynamic threshold..."""`).
Reference: [Google Python Style Guide #383 functions and methods](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods)

Reply:
https://github.com/EE-Ind-Stud-Group/blink-detection/pull/5/commits/650a731c04d47618cd466ac42b167be02867cee8